sanic/types/shared_ctx.py

```diff
@@ -0,0 +1,55 @@
+import os
+
+from types import SimpleNamespace
+from typing import Any, Iterable
+
+from sanic.log import Colors, error_logger
+
+
+class SharedContext(SimpleNamespace):
+    SAFE = ("_lock",)
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self._lock = False
+
+    def __setattr__(self, name: str, value: Any) -> None:
+        if self.is_locked:
+            raise RuntimeError(
+                f"Cannot set {name} on locked SharedContext object"
+            )
+        if not os.environ.get("SANIC_WORKER_NAME"):
+            to_check: Iterable[Any]
+            if not isinstance(value, (tuple, frozenset)):
+                to_check = [value]
+            else:
+                to_check = value
+            for item in to_check:
+                self._check(name, item)
+        super().__setattr__(name, value)
+
+    def _check(self, name: str, value: Any) -> None:
+        if name in self.SAFE:
+            return
+        try:
+            module = value.__module__
+        except AttributeError:
+            module = ""
+        if not any(
+            module.startswith(prefix)
+            for prefix in ("multiprocessing", "ctypes")
+        ):
+            error_logger.warning(
+                f"{Colors.YELLOW}Unsafe object {Colors.PURPLE}{name} "
+                f"{Colors.YELLOW}with type {Colors.PURPLE}{type(value)} "
+                f"{Colors.YELLOW}was added to shared_ctx. It may not "
+                "not function as intended. Consider using the regular "
+                f"ctx. For more information, please see ____.{Colors.END}"
+            )
```

Comment:
Not sure of a better way to do this. If anyone has some thoughts, happy to hear.

Reply:
I think this is fine for now and don't think we need to block the merge for a possible better way.

Reply:
I did play with the whitelist idea, but it really gets messy and out of control quick. Just take a look at what the `multiprocessing` module is doing and I think you will see why that seems like a bad idea.

https://github.com/python/cpython/blob/main/Lib/multiprocessing/__init__.py#L22
https://github.com/python/cpython/blob/main/Lib/multiprocessing/managers.py#L1211